nntrainer/tensor/cache_loader.cpp

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2022 Jiho Chu <jiho.chu@samsung.com>
+ *
+ * @file   cache_loader.cpp
+ * @date   10 Nov 2022
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Jiho Chu <jiho.chu@samsung.com>
+ * @bug    No known bugs except for NYI items
+ * @brief  Cache loader class
+ *
+ */
+
+#include "cache_loader.h"
+#include "task.h"
+#include "task_executor.h"
+
+#include <cache_pool.h>
+#include <climits>
+#include <cstdint>
+#include <exception>
+#include <memory>
+#include <nntrainer_log.h>
+
+namespace nntrainer {
+
+CacheLoader::CacheLoader(std::shared_ptr<CachePool> cache_pool) :
+  pool(cache_pool),
+  task_executor(nullptr) {}
+
+CacheLoader::~CacheLoader() {
+  if (task_executor)
+    delete task_executor;
+}
+
+void CacheLoader::init() {
+  if (task_executor)
+    return;
+
+  task_executor = new TaskExecutor(pool->getName());
+}
+
+void CacheLoader::finish() {
+  delete task_executor;
+  task_executor = nullptr;
+}
+
+void CacheLoader::load(unsigned int order) { pool->loadExec(order); }
```

Comment:
it calls ```pool->loadExec(order)```, and in ```loadExec(order)```, start searching the ```elems```,  ```CacheElems``` and check if the ```elem``` in ```elems``` has same execution order as ```order```. It's kind of something to be done definitely but I'm thinking, it only needs to be done first iteration. and rest of training processes, we do not need to compute which elem needs be loaded  at specific ```order```. Becuase it will be always same once we compute. 
So how about ```CachePool``` has those scheduled information according to execution order.. and if it is set, then just load. I think we do not validate either.

Reply:
Thanks for sharing idea.
It's implemented in #2055, and you may be already checked while reviweing #2055.